File: scraper.py
import argparse
import os
from bs4 import BeautifulSoup
import requests

def parse_arguments():
    parser = argparse.ArgumentParser(description="Bass Guitar Scraper")
    parser.add_argument("urls", nargs="*", default=os.getenv("SCRAPE_URLS", "").split(";"),
                        help="URLs to scrape, separated by space")
    parser.add_argument("--selector", "-s", default=os.getenv("SELECTOR", "h3.bass-guitar-name"),
                        help="CSS selector to use for scraping")
    args = parser.parse_args()
    return args

def scrape_bass_guitar_info(urls, selector):
    for url in urls:
        response = requests.get(url)
        if response.status_code != 200:
            print("Failed to fetch the webpage")
            continue
        
        soup = BeautifulSoup(response.content, 'html.parser')
        elements = soup.select(selector)
        # Extract information based on the provided selector
        for element in elements:
            print(element.text)

if __name__ == "__main__":
    args = parse_arguments()
    scrape_bass_guitar_info(args.urls, args.selector)

File: scraper.rs
use clap::{App, Arg};
use dotenv::dotenv;
use std::env;
use reqwest::blocking::get;
use select::document::Document;
use select::predicate::Predicate;
use select::predicate::Class;

fn parse_arguments() {
    dotenv().ok();

    let matches = App::new("Bass Guitar Scraper")
        .arg(Arg::with_name("urls")
            .value_name("URLs")
            .multiple(true)
            .help("URLs to scrape, separated by space")
            .default_value(&env::var("SCRAPE_URLS").unwrap_or_default())
            .use_delimiter(true))
        .arg(Arg::with_name("selector")
            .short("s")
            .long("selector")
            .value_name("SELECTOR")
            .help("CSS selector to use for scraping")
            .default_value(&env::var("SELECTOR").unwrap_or("h3.bass-guitar-name".to_string())))
        .get_matches();

    let urls: Vec<&str> = matches.values_of("urls").unwrap_or_default().collect();
    let selector = matches.value_of("selector").unwrap_or("h3.bass-guitar-name");
    scrape_bass_guitar_info(&urls, selector);
}

fn scrape_bass_guitar_info(urls: &[&str], selector: &str) {
    for url in urls {
        let response = get(url);
        match response {
            Ok(res) => {
                if res.status().is_success() {
                    let document = Document::from_read(res).unwrap();
                    for node in document.find(Class(selector)) {
                        println!("{}", node.text());
                    }
                } else {
                    println!("Failed to fetch the webpage");
                }
            }
            Err(e) => println!("Error: {}", e),
        }
    }
}

fn main() {
    parse_arguments();
}
